### PR TITLE
Switch to Alcotest for unit testing

### PR DESCRIPTION
**Pull request by mjambon**
Uses https://github.com/returntocorp/pfff/pull/468

It looks like this when successful:

Screenshot_20210908_051534

And it looks like this in case of an error:

Screenshot_20210908_052153

PR checklist:
- [x] documentation is up to date
- [x] changelog is up to date

### CODE REVIEW

File: semgrep-core/src/testutil/Testutil.mli [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+   value as its output, it's just easier to fail with 'assert false'.
+
+   In any case, Alcotest will capture the output (stdout, stderr) of each
+   test and put it in its own file so we can consult it later. Don't
+   hesitate to log a lot during the execution of the test.
+*)
+type test = string * (unit -> unit)
+
+(*
+   This extends the name of each test by adding a prefix, using '.'
```
Comment (aryx) [inline semgrep-core/src/testutil/Testutil.mli]: I thought the separator was '>' in the end, not '.'.

**Review by aryx — APPROVED** [review]

**Comment by emjin** [comment]
Looks great! Can we merge?

**Review by emjin — APPROVED** [review]

File: semgrep-core/src/testutil/Testutil.mli (hunk 1 above)
Comment (mjambon) [reply semgrep-core/src/testutil/Testutil.mli]: thx! I updated the doc.

**Comment by IagoAbal** [comment]
Nice!

**Pull request merged**